openshift/hack/rebase_upstream.sh

```
@@ -46,7 +46,13 @@ if [[ $? -ne 0 ]]; then
   exit 1
 fi
 git merge "$remote_branch"
-git checkout -b "$version"-rebase-"$rebase_branch" || { echo "Expected branch $version-rebase-$rebase_branch to not exist, delete and retry."; popd ; exit 1; }
+rebase_work_branch="$version"-rebase-"$rebase_branch"
+if git show-ref --verify --quiet refs/heads/"$rebase_work_branch"; then
+  echo "Branch $rebase_work_branch already exists, reusing it (assuming any pre-staged commits, e.g. cherry-picks, are intentional)."
+  git checkout "$rebase_work_branch"
+else
+  git checkout -b "$rebase_work_branch" || { echo "Failed to create branch $rebase_work_branch."; popd ; exit 1; }
```

Shellcheck (0.11.0)

[warning] Use 'popd ... || exit' or 'popd ... || return' in case popd fails.

(SC2164)